### PR TITLE
Added cold expanding beam benchmark example (with space charge).

### PR DESCRIPTION
**Pull request by cemitch99**
Simple space charge benchmark, consisting of a 1 nC electron beam coasting in a drift with kinetic energy of 250 MeV.  The beam distribution is a uniformly-populated ellipsoid, and the drift distance is chosen such that the beam should expand to twice its original size.

- [x] add in `examples/CMakeLists.txt`
- [x] add in `docs/source/usage/examples.rst`
- [x] add `examples/expanding/run_expanding.py`

### CODE REVIEW

**Comment by ax3l** [comment]
Thanks a lot! :sparkles: 

The test seems to still pass, is the tolerance too wide?

**Comment by cemitch99** [comment]
> Thanks a lot! ✨
> 
> The test seems to still pass, is the tolerance too wide?

Let me look into this.  I verified that, with space charge off, the beam moments do not change significantly.  The test should pass only if sigx, sigy, and sigt all are doubled, so the relative tolerance must be ~1(?)

File: examples/expanding_beam/README.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.. _examples-expanding:
+
+Expanding Beam in Free Space
+=============================
```
Comment (ax3l) [inline examples/expanding_beam/README.rst]:
```suggestion
Expanding Beam in Free Space
============================
```

**Comment by ax3l** [comment]
The tolerances are just absolute/relative to the numeric values you give for the final distribution (not the initial ones). So keep them small.

File: examples/expanding_beam/analysis_expanding.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+print(f"  sigx={sigx:e} sigy={sigy:e} sigt={sigt:e}")
+print(
+    f"  emittance_x={emittance_x:e} emittance_y={emittance_y:e} emittance_t={emittance_t:e}"
+)
+
+atol = 1.0  # a big number
+rtol = num_particles**-0.5  # from random sampling of a smooth distribution
+print(f"  rtol={rtol} (ignored: atol~={atol})")
+
+assert np.allclose(
```
Comment (ax3l) [inline examples/expanding_beam/analysis_expanding.py:103]: If emittance are close to zero, maybe do a relative tolerance for the beam widths and an absolute one for the emitttance (two separate asserts)

File: examples/expanding_beam/README.rst [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+This example can be run as a Python script (``python3 run_expanding.py``) or with an app with an input file (``impactx input_expanding.in``).
+Each can also be prefixed with an `MPI executor <https://www.mpi-forum.org>`__, such as ``mpiexec -n 4 ...`` or ``srun -n 4 ...``, depending on the system.
+
+.. tab-set::
+
+   .. tab-item:: Python Script
+
+      .. literalinclude:: run_expanding.py
+         :language: python3
+         :caption: You can copy this file from ``examples/expanding/run_expanding.py``.
```
Comment (ax3l) [inline examples/expanding_beam/README.rst:30]: I think this file is not yet added in the pull request?

File: examples/expanding_beam/input_expanding.in [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+drift1.type = drift
+drift1.ds = 6.0
+
+
+###############################################################################
+# Algorithms
+###############################################################################
+algo.particle_shape = 2
+algo.space_charge = true
```
Comment (ax3l) [inline examples/expanding_beam/input_expanding.in:35]: We can now add the desired grid resolution for our beam #240, e.g.,
```suggestion
algo.space_charge = true

amr.n_cell = 40 40 32
geometry.prob_relative = 1.0
```
passed for me, using only `lattice.nslice = 30` :)

File: examples/expanding_beam/analysis_expanding.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+assert num_particles == len(final)
+
+print("Initial Beam:")
+sigx, sigy, sigt, emittance_x, emittance_y, emittance_t = get_moments(initial)
+print(f"  sigx={sigx:e} sigy={sigy:e} sigt={sigt:e}")
+print(
+    f"  emittance_x={emittance_x:e} emittance_y={emittance_y:e} emittance_t={emittance_t:e}"
+)
+
+atol = 1.0  # a big number
```
Comment (ax3l) [inline examples/expanding_beam/analysis_expanding.py]: #248:
```suggestion
atol = 0.0  # ignored
```

File: examples/expanding_beam/analysis_expanding.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+print("")
+print("Final Beam:")
+sigx, sigy, sigt, emittance_x, emittance_y, emittance_t = get_moments(final)
+print(f"  sigx={sigx:e} sigy={sigy:e} sigt={sigt:e}")
+print(
+    f"  emittance_x={emittance_x:e} emittance_y={emittance_y:e} emittance_t={emittance_t:e}"
+)
+
+atol = 1.0  # a big number
```
Comment (ax3l) [inline examples/expanding_beam/analysis_expanding.py]: #248:
```suggestion
atol = 0.0  # ignored
```

File: examples/expanding_beam/input_expanding.in [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+beam.muxpx = 0.0
+beam.muypy = 0.0
+beam.mutpt = 0.0
+
+
+###############################################################################
+# Beamline: lattice elements and segments
+###############################################################################
+lattice.elements = drift1
+lattice.nslice = 100
```
Comment (ax3l) [inline examples/expanding_beam/input_expanding.in]: Sufficient :tada: 
```suggestion
lattice.nslice = 30
```

File: examples/expanding_beam/README.rst (hunk 3 above)
Comment (ax3l) [reply examples/expanding_beam/README.rst:30]: I added it now :)

**Comment by ax3l** [comment]
Hi @cemitch99 , I finalized this PR:
- added a Python inputs set
- increase the relative tolerance slightly by 1.5x
- disable the test, so this can be merged without failing, and enable it again in #162

**Review by ax3l — APPROVED** [review]

**Pull request merged**